### PR TITLE
Create Quantpendia and Smasher

### PR DESCRIPTION
**Pull request by bug-sam**
## Related Issues:

#3 
#9

This time I added a spellchecker extension to vscode so hopefully there will be no small spelling errors!!

### CODE REVIEW

File: smasher.md [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+    - creates second test Experiment
+    - creates second test Sample
+    - associates the test Sample to the ComputationalResult and Experiment
+    - creates second test ComputedFile and associates it to the second test Sample
+    - creates test Dataset with the two Experiments and Samples
+    - associates the Dataset to the test ProcessorJob
+- runs the smasher pipeline
+- asserts that the first Sample's ComputedFile was added to the job context's list of unsmashable files
+
+### `test_qn_targets_only`
```
Comment (jaclyn-taroni) [inline smasher.md:119]: Is this the right header? I expect not based on the text below but also may be confused.

**Review by jaclyn-taroni — APPROVED** [review]
LGTM with the exception of one header I think might be wrong. We should have @kurtwheeler take a look though!

File: smasher.md (hunk 1 above)
Comment (kurtwheeler) [reply smasher.md:119]: I think the test had the wrong name. Perhaps when it was written whoever did so mixed up QN and quant?

File: smasher.md [hunk 2]
```diff
@@ -0,0 +1,593 @@
...
+    - asserts that the number of experiments is set to 1
+    - asserts that `aggregate_by` is not set in the metadata
+
+### `test_no_smash_dupe`
+
+Tests the case where two Samples are associated to the same ComputedFile.
+
+** In this test there are additional places where test ComputationalResults are created.
+I think they don't actually need to be created -
+hey were kind of confusing to me and removing them or adding comments for their would make the test less confusing
```
Comment (kurtwheeler) [inline smasher.md]:
```suggestion
they were kind of confusing to me and removing them or adding comments for their would make the test less confusing
```

Can you make a refinebio issue to do this?

File: smasher.md [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+    - runs the smasher pipeline
+    - asserts that the output file exists
+    - removes the output file
+    - asserts that the final output frame has 1 columns
+- aggregate by "ALL"
+    - gets the test Dataset and sets `aggregate_by` to be "ALL"
+    - resets the test ProcessorJob start and end times
+    - runs the smasher pipeline
+    - asserts that the output file exists
+    - asserts that the final output frame has 1 columns
```
Comment (kurtwheeler) [inline smasher.md]: For this and EXPERIMENT, shouldn't there still be two columns, just in different matrices?

File: smasher.md [hunk 4]
```diff
@@ -0,0 +1,593 @@
...
+- asserts that the job ran without any failures
+
+## Compendia Tests
+
+### `test_call_create`
+
+Tests the `create_compendia` call command.
+
+test steps:
+- rerouts stderr and stdout
```
Comment (kurtwheeler) [inline smasher.md]:
```suggestion
- reroutes stderr and stdout
```

File: smasher.md [hunk 5]
```diff
@@ -0,0 +1,593 @@
...
+- asserts that the `create_compendia` call command raises an exception
+- resets stderr and stdout
+
+### `test_fetch_create`
+
+Tests the `fetch_compendia` call command.
+
+test steps:
+- rerouts stderr and stdout
```
Comment (kurtwheeler) [inline smasher.md]:
```suggestion
- reroutes stderr and stdout
```

**Review by kurtwheeler — APPROVED** [review]
Sweet, this looks great! I had a couple minor comments but this LGTM!

File: smasher.md (hunk 3 above)
Comment (bug-sam) [reply smasher.md]: I was just wrong for "ALL" it should be 2.

I think for experiment, there would be 2 files, but `job_context["final_frame"]` just gets overwritten for each key that's smashed. So I think it's only checking for the second Experiment which would only have 1 column.

File: smasher.md (hunk 2 above)
Comment (bug-sam) [reply smasher.md]: created https://github.com/AlexsLemonade/refinebio/issues/2650!

File: smasher.md (hunk 1 above)
Comment (bug-sam) [reply smasher.md:119]: added a comment to change the name.
also made it a part of https://github.com/AlexsLemonade/refinebio/issues/2650

**Pull request merged**